### PR TITLE
🌐 #313: add function to translate namespace

### PR DESCRIPTION
**Pull request by samrichca**
### Consent UI

- add a function that translates an entire namespace
- refactor to use translateNamespace in places where the same dictionary is translated multiple times

## PR Readiness Checklist

- [x] "Expected Outcome(s)" in ticket have been met
- [x] Ticket number included in PR title
- [x] Connected ticket to PR
- [x] Labels added to PR for service name (`consent-api`, `data-mapper`, etc...), type (`chore`, `documentation`, etc...), status (`draft`, `on-hold`, etc...) **if applicable**
- [x] Manual testing completed
- [x] Builds locally without errors or warnings
- [x] Tests are updated (if required) and passing
- [ ] PR feedback has been addressed **if applicable**
- [x] New environment variables added to `.env.schema` files, `README.md`
- [x] Added copyrights to new files

### CODE REVIEW

**Comment by samrichca** [comment]
@anncatton good point about enforcing server only for translations, we should get on that

I'll update the places where I'm not fetching the entire dictionary.

File: docs/translations.md [hunk 1]
```diff
@@ -55,7 +55,7 @@ const frenchCommonDictionary = { ... } satisfies CommonDictionary;
```
[the remark this answers is not in the document]
Comment (samrichca) [reply docs/translations.md:54]: updated readme and added jsdoc comments

File: docs/translations.md [hunk 2]
```diff
@@ -74,22 +75,46 @@ const translatedString = translate('footer', 'about');
...
 ```typescript
-const translate = getTranslation('fr');
+const { translate } = getTranslation('fr');
 const translatedString = translate('footer', 'copyright', { year: '2023' });
 // => '© 2023 Ontario Hereditary Cancer Research Network. Tous droits réservés.'
 ```
 
+#### Translating a Namespace
+
+Translate an entire namespace at once when you want to pass down many translations from a server component to a client component, e.g. for forms.
```
Comment (anncatton) [inline docs/translations.md:85]: this is great, thanks for adding. One smol request, can you add a note about the "why" here - we are keeping the translations on the server side, so we can't access the functions in these client components. `translateNamespace` is a convenience function to retrieve all necessary translations in a server parent, and pass them to a child client component.

File: apps/consent-ui/src/i18n/utils/getTranslation.ts [hunk 3]
```diff
@@ -56,7 +56,18 @@ export const replaceParams = (
 // TODO: is there a way to enforce this function for server side use only?
 export const getTranslation: GetTranslation = (language) => {
 	const dictionary = dictionaries[language];
-	return (namespace, key, params) => {
+
+	/**
```
Comment (anncatton) [inline apps/consent-ui/src/i18n/utils/getTranslation.ts:60]: adding jsdocs is great. it looks though like Intellisense is not able to pick up the comments when they are nested inside a function like this. Poking around I didn't see any satisfactory solution...i think just as a little helper here, we can add a simple jsdoc to `getTranslation`, maybe something like:
```
/**
 * @param language ValidLanguage
 * @exports {translate, translateNamespace}
 * /
```

**Review by anncatton — COMMENTED** [review]
good stuff. just a couple minor comments and then should be good to go!

**Comment by anncatton** [comment]
> @anncatton good point about enforcing server only for translations, we should get on that
> 
> I'll update the places where I'm not fetching the entire dictionary.

Forgot to add in my other review, i'll make a separate ticket for the server-only setup and we can discuss at planning next week

**Comment by anncatton** [comment]
Created a ticket for the server-only task https://github.com/OHCRN/platform/issues/422

**Review by anncatton — APPROVED** [review]

**Pull request merged**